File: courses/codes/HW7_recursion_CCC1_homework_solutions/Q1.py
def grow(s, k):
    if k == 0:
        return s
    r = []
    for i in s:
        r.append(i + "0")
        if i[-1] == '0':
            r.append(i + "1")

    return grow(r, k-1)


d = int(input())
zeroes = grow(["0"], d-1)
ones = grow(["1"], d-1)

for a in zeroes + ones:
    print(a)

File: courses/codes/cclv1-jr-class/HW7_recursion_CCC1_homework_solutions/Q1.java
import java.util.*;

public class Q1 {

    public static List<String> grow(List<String>s, int k) {
        if (k == 0) {
            return s;
        }
        List<String> r = new LinkedList<String>();
        for (String i: s) {
            r.add(i + "0");
            if (i.charAt(i.length() - 1) == '0') {
                r.add(i + "1");
            }
        }
        return grow(r, k-1);
    }

    public static void main(String args[]) {
        Scanner scan = new Scanner(System.in);
        int d = scan.nextInt();
        List<String> zero = new LinkedList<String>();
        zero.add("0");
        List<String> zeroes = grow(zero, d-1);
        List<String> one = new LinkedList<String>();
        one.add("1");
        List<String> ones = grow(one, d-1);

        for (String a: zeroes) {
            System.out.println(a);
        }

        for (String a: ones) {
            System.out.println(a);
        }
        scan.close();
    }
}
